src/Umbraco.Web/UmbracoContext.cs

```diff
@@ -37,7 +38,8 @@ public class UmbracoContext : DisposableObjectSlim, IDisposeOnRequestEnd
             IEnumerable<IUrlProvider> urlProviders,
             IEnumerable<IMediaUrlProvider> mediaUrlProviders,
             IGlobalSettings globalSettings,
-            IVariationContextAccessor variationContextAccessor)
+            IVariationContextAccessor variationContextAccessor,
+            IPublishedSnapshotAccessor publishedSnapshotAccessor)
```

Comment:
This is a circular refernce. The IPublishedSnapshotAccessor depends on the UmbracoContext, the default implementation is UmbracoContextPublishedSnapshotAccessor. The UmbracoContext should not depend on or expose this service.

Reply:
Removed